sdk/schemaregistry/azure-data-schemaregistry-jsonschema/src/main/java/com/azure/data/schemaregistry/jsonschema/SchemaRegistryJsonSchemaSerializerBuilder.java

```diff
@@ -85,6 +87,19 @@ public SchemaRegistryJsonSchemaSerializerBuilder schemaRegistryClient(
         return this;
     }
 
+    /**
+     * The {@link SchemaRegistryClient} to use to interact with the Schema Registry service.
+     *
+     * @param schemaRegistryClient The {@link SchemaRegistryClient}.
+     *
+     * @return updated {@link SchemaRegistryJsonSchemaSerializerBuilder} instance.
+     */
+    public SchemaRegistryJsonSchemaSerializerBuilder schemaRegistryClient(
```

Comment:
On the `buildSerializer` method, we have this validation. We should check if either the sync or the async client is present. 
- if both are null, throw exception
- if both are non-null, use respective clients for sync and async calls
- if only sync client is non-null, only sync calls work and async calls throw
- if only async client is non-null, both sync and async calls should work (using sync-over-async)

```java
if (Objects.isNull(schemaRegistryAsyncClient)) {
     throw LOGGER.logExceptionAsError(new NullPointerException("'schemaRegistryAsyncClient' cannot be null."));
}
```